dace/sdfg/validation.py

```diff
@@ -1156,3 +1167,14 @@ def validate_memlet_data(memlet_data: str, access_data: str) -> bool:
     memlet_tokens = memlet_data.split('.')
     mem_root = '.'.join(memlet_tokens[:len(access_tokens)])
     return mem_root == access_data
+
+
+def _no_writes_to_scalars_or_arrays_on_interstate_edges(sdfg: 'dace.sdfg.SDFG'):
+    from dace.sdfg import InterstateEdge
+    for edge, graph in sdfg.all_edges_recursive():
```

Comment:
There's also `all_interstate_edges` - though as the previous comment states, this should imo be changed to go over a single control flow region only and called per control flow region.